clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -24,8 +24,14 @@ using namespace clang;
 using namespace cir;
 
 unsigned CIRGenTypes::ClangCallConvToCIRCallConv(clang::CallingConv CC) {
-  assert(CC == CC_C && "No other calling conventions implemented.");
-  return cir::CallingConv::C;
+  switch (CC) {
+  case CC_C:
+    return cir::CallingConv::C;
+  case CC_OpenCLKernel:
+    return CGM.getTargetCIRGenInfo().getOpenCLKernelCallingConv();
```

Comment:
Since we do not support `MissingFeatures::setCallingConv()`, this won't reflect on the final IR, but it is required for tests to compile.

Reply:
@sitio-couto please review this PR